src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -1246,7 +1247,7 @@ private void AssignUnscheduledRequestsWithMaxWaitingRequests2(List<ScheduleRespo
         private void AssignUnscheduledRequestsFIFO(List<ScheduleResponse> responses, HashSet<int> idleNodes)
         {
             // Assign requests on a first-come/first-serve basis
-            foreach (int nodeId in idleNodes)
+            foreach (SchedulableRequest unscheduledRequest in _schedulingData.UnscheduledRequestsWhichCanBeScheduled)
```

Comment:
What's the motivation for the switch? Document in a comment.

Reply:
Enumerating over `UnscheduledRequestsWhichCanBeScheduled` is more expensive than enumerating over `idleNodes` (`HashSet<int>`). To enumerate `UnscheduledRequestsWhichCanBeScheduled`, we allocate a new enumerator each time (from the yield return) and the linked list enumeration is slightly slower too. I can add a comment to clarify if we want to keep this change.